src/core/data-structures/weakref.d.ts

```diff
@@ -0,0 +1,7 @@
+export {};
+
+declare global {
+interface Window {
+  WeakRef?: typeof WeakRef;
```

Comment:
Is WeakRef not already globally available? Do we need to add `dom` to the tsconfig or something?

Reply:
Correct. This pattern is used for a handful of things (esp. Observer classes and some native streaming classes) that are global, but we access via `Window` so our polyfill is available if native is not.